api/well_inventory.py

```diff
@@ -0,0 +1,203 @@
+from typing import Optional, Set
+
+from fastapi import APIRouter, UploadFile, File
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel, ValidationError
+from shapely import Point
+from sqlalchemy import select
+
+from constants import SRID_UTM_ZONE_13N, SRID_UTM_ZONE_12N, SRID_WGS84
+from core.dependencies import session_dependency
+from core.enums import ContactType, Role, ElevationMethod
+from db import (
+    Group,
+    ThingIdLink,
+    GroupThingAssociation,
+    Location,
+    LocationThingAssociation,
+)
+from db.thing import Thing
+from services.util import transform_srid
+
+router = APIRouter(prefix="/well-inventory-csv")
+
+
+class WellInventoryRow(BaseModel):
+    project: str
+    well_name_point_id: str
+    site_name: str
+    date_time: datetime
+    field_staff: str
+    utm_easting: float
+    utm_northing: float
+    utm_zone: int
+    elevation_ft: float
+    elevation_method: ElevationMethod
+    measuring_point_height_ft: float
+
+    # Optional lexicon fields
+    contact_role: Optional[Role] = None
+    contact_type: Optional[ContactType] = None
+
+
+@router.post("")
+async def well_inventory_csv(session: session_dependency, file: UploadFile = File(...)):
+    if not file.content_type.startswith("text/csv") or not file.filename.endswith(
+        ".csv"
+    ):
+        return JSONResponse(status_code=400, content={"error": "Unsupported file type"})
+
+    content = await file.read()
+    if not content:
+        return JSONResponse(status_code=400, content={"error": "Empty file"})
+    try:
+        text = content.decode("utf-8")
+    except Exception:
+        return JSONResponse(status_code=400, content={"error": "File encoding error"})
+    reader = csv.DictReader(StringIO(text))
+    rows = list(reader)
+    if not rows:
+        return JSONResponse(status_code=400, content={"error": "No data rows found"})
+    validation_errors = []
+    wells = []
+    models = []
+    seen_ids: Set[str] = set()
+    for idx, row in enumerate(rows):
+        try:
+            well_id = row.get("well_name_point_id")
+            if not well_id:
+                raise ValueError("Field required")
+            if well_id in seen_ids:
+                raise ValueError("Duplicate value for well_name_point_id")
+            seen_ids.add(well_id)
+            model = WellInventoryRow(**row)
+            models.append(model.model_dump())
+
+        except ValidationError as e:
+            for err in e.errors():
+                validation_errors.append(
+                    {
+                        "row": idx + 1,
+                        "field": err["loc"][0],
+                        "error": f"Value error, {err['msg']}",
+                    }
+                )
+        except ValueError as e:
+            validation_errors.append(
+                {"row": idx + 1, "field": "well_name_point_id", "error": str(e)}
+            )
+
+    def convert_f_to_m(r):
+        return r * 0.3048
```

Comment:
When BDMS-221 and BDMS-227 are complete and merged this function will be in services/util.py and can be removed from here then imported